drivers/ethernet/eth_stm32_hal.c

```diff
@@ -340,8 +341,11 @@ static void eth0_iface_init(struct net_if *iface)
 
 	heth->Init.MACAddr = dev_data->mac_addr;
 
-	if (HAL_ETH_Init(heth) != HAL_OK) {
-		SYS_LOG_ERR("HAL_ETH_Init failed\n");
+	hal_ret = HAL_ETH_Init(heth);
+
+	if (hal_ret != HAL_OK) {
+		SYS_LOG_ERR("HAL_ETH_Init failed: %d\n", hal_ret);
+		return;
 	}
```

Comment:
Does it make sense to move this before we call k_thread_create?

Reply:
Yeah so we avoid having thread being created and left unused, this is right.